src/backend/api/redis.py

```diff
@@ -21,15 +23,17 @@
 CACHE_EXPIRED_TIME = 24 * 3600  # 24 hours
 
 
-def get_mirrors_from_cache(
+async def get_mirrors_from_cache(
         key: AnyStr,
 ) -> Optional[List[MirrorData]]:
     """
     Get a cached list of mirrors for specified IP
     """
     key = str(key)
     redis_engine = RedisEngine.get_instance()
-    mirrors_string = redis_engine.get(key)
+    mirrors_string = await redis_engine.get(key)
+    await redis_engine.close()
```

Comment:
Can we move the two lines to a separate function?

Reply:
The engine closing?  I don't see why...